File: Chaarsu/Controllers/UserController.cs
using Chaarsu.Repository.Interface;
using Chaarsu.Repository.SPRepository;
using Chaarsu.Repository.UnitofWork;
using Chaarsu.Models;
using Chaarsu.Repository.GRepository;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace Chaarsu.Controllers
{
    public class UserController : Controller
    {
        private GenericRepository<USER> _USER;
        private GenericRepository<USER_ADDRESSES> _UserAddresses;
        private GenericRepository<ORDER_STATUSES> _OrderStatues;
        private readonly IUnitOfWork _unitOfWork;
        private readonly SpRepository _sp;

        public UserController()
        {
            _unitOfWork = new UnitOfWork();
            _sp = new SpRepository();
        }

        // GET: User
        [HttpGet]
        public ActionResult Profile()
        {
            if (Session["UserSession"] != null)
            {
                return View();
            }
            else
            {
                return RedirectToAction("SignIn", "Accounts", new { area = "" });
            }           
        }

        // GET: User
        [HttpGet]
        public ActionResult TrackOrder()
        {
            if (Session["UserSession"] != null)
            {
                return View();
            }
            else
            {
                return RedirectToAction("SignIn", "Accounts", new { area = "" });
            }
        }


        [HttpGet]
        public JsonResult GetUserDetail()
        {
            try
            {
                _USER = new GenericRepository<USER>(_unitOfWork);
                var UserId = Convert.ToInt32(Session["UserSession"].ToString());
                var entity = _USER.Repository.Get(x => x.USER_ID == UserId);
                if (entity != null)
                {
                    return Json(entity, JsonRequestBehavior.AllowGet);
                }
                else
                {
                    return Json("Invalid", JsonRequestBehavior.AllowGet);
                }
            }
            catch (Exception ex)
            {
                return Json(new { Status = false, RetMessage = ex.Message });
            }
        }

        [HttpGet]
        public JsonResult GetAddressList()
        {
            try
            {
                _UserAddresses = new GenericRepository<USER_ADDRESSES>(_unitOfWork);
                var UserId = Convert.ToInt32(Session["UserSession"].ToString());
                var entity = _UserAddresses.Repository.GetAll(x => x.USER_ID == UserId).ToList();
                if (entity != null)
                {
                    return Json(entity, JsonRequestBehavior.AllowGet);
                }
                else
                {
                    return Json("Invalid", JsonRequestBehavior.AllowGet);
                }
            }
            catch (Exception ex)
            {
                return Json(new { Status = false, RetMessage = ex.Message });
            }
        }

        [HttpGet]
        public JsonResult UpdateUserDetail(string username, string email, string mobile, string address)
        {
            try
            {
                var UserId = Convert.ToInt32(Session["UserSession"].ToString());
                _USER = new GenericRepository<USER>(_unitOfWork);
                var emailExist = _USER.Repository.Get(x => x.USER_ID != UserId && x.EMAIL == email);
                var mobileExist = _USER.Repository.Get(x => x.USER_ID != UserId && x.MOBILE_NO == mobile);             
                if (emailExist != null)
                {
                    return Json(new { Status = false, RetMessage = "Email already exist" }, JsonRequestBehavior.AllowGet);
                }
                else if (mobileExist != null)
                {
                    return Json(new { Status = false, RetMessage = "Mobile number already exist" }, JsonRequestBehavior.AllowGet);
                }
                else
                {
                    var entity = _USER.Repository.Get(x => x.USER_ID == UserId);
                    entity.USERNAME = username;
                    entity.EMAIL = email;
                    entity.MOBILE_NO = mobile;
                    entity.ADDRESS = address;
                    entity.UPDATED_BY = UserId;
                    entity.UPDATED_ON = DateTime.Now;
                    _USER.Repository.Update(entity);
                    return Json(new { Status = true, RetMessage = "Your account detail updated successfully" }, JsonRequestBehavior.AllowGet);
                }
                
            }
            catch (Exception ex)
            {
                return Json(new { Status = false, RetMessage = ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }

        [HttpGet]
        public JsonResult ChangePassword(string OldPassword, string NewPassword)
        {
            try
            {
                _USER = new GenericRepository<USER>(_unitOfWork);
                var UserId = Convert.ToInt32(Session["UserSession"].ToString());
                var entity = _USER.Repository.Get(x => x.USER_ID == UserId && x.PASSWORD == OldPassword);
                if (entity != null)
                {
                   entity.PASSWORD = NewPassword;
                   entity.UPDATED_BY = UserId;
                   entity.UPDATED_ON = DateTime.Now;
                   _USER.Repository.Update(entity);
                   Session.RemoveAll();
                   Session.Abandon();
                   return Json(new { Status = true, RetMessage = "Your password changed successfully" }, JsonRequestBehavior.AllowGet);
                }
                else
                {
                   return Json(new { Status = false, RetMessage = "Your old password is invalid" }, JsonRequestBehavior.AllowGet);
                }
                
            }
            catch (Exception ex)
            {
                return Json(new { Status = false, RetMessage = ex.Message }, JsonRequestBehavior.AllowGet);
            }
        }

        [HttpGet]
        public JsonResult GetAllOrders(int PageIndex, int PageSize, string SortColumn, string SortOrder, string SearchText, int OrderStatusId)
        {
            try
            {
                if (Session["UserSession"] == null)
                {
                    return Json("AccessDenied", JsonRequestBehavior.AllowGet);
                }
                else
                {
                    var CustomerId = Convert.ToInt32(Session["UserSession"].ToString());
                    var orders = _sp.GetAllOrders(PageIndex, PageSize, SortColumn, SortOrder, SearchText, OrderStatusId, CustomerId);
                    if (orders == null)
                    {
                        return Json("AccessDenied", JsonRequestBehavior.AllowGet);
                    }
                    else
                    {

                        return Json(orders, JsonRequestBehavior.AllowGet);
                    }
                }
            }
            catch (Exception ex)
            {
                return Json(new { Status = false, RetMessage = ex.Message });
            }
        }

    }
}

File: Chaarsu/ngControllers/UserController.js
var objCommon = new Common();
suchiapp.controller("UserController", function ($scope) {
    $scope.loader = true;
    $scope.GetAllOrders = function (orderStatusId) {
        $("body").css({ "opacity": "0.5" });
        $scope.loader = true;
        $scope.pageIndex = 1;
        $scope.pageSize = 20;
        $scope.sortColumn = "";
        $scope.sortOrder = "";
        $scope.searchText = "";
        $scope.orderStatusId = orderStatusId;
        var data = {
            PageIndex: $scope.pageIndex,
            PageSize: $scope.pageSize,
            SortColumn: $scope.sortColumn,
            SortOrder: $scope.sortOrder,
            SearchText: $scope.searchText,
            OrderStatusId: $scope.orderStatusId
        };
        objCommon.AjaxCall("User/GetAllOrders", $.param(data), "GET", true, function (data) {
            if (data == "AccessDenied") {
                objCommon.ShowMessage("Access Denied Your Session is Time Out.", "error");
            }
            else {
                $scope.Orders = data;
            }
            $scope.$apply();
        }, null, "Error while getting all orders, Please try again.", $scope);
        $scope.AccountDetail();
    }

    $scope.AccountDetail = function () {
        objCommon.AjaxCall("User/GetUserDetail", $.param({}), "GET", true, function (data) {
            if (data == "Invalid") {
                objCommon.ShowMessage("Access Denied Your Session is Time Out.", "error");
            }
            else {
                $scope.username = data.USERNAME;
                $scope.email = data.EMAIL;
                $scope.mobile = data.MOBILE_NO;
                $scope.address = data.ADDRESS;
            }
            $("body").css({ "opacity": "1" });
            $scope.loader = false;
            $scope.$apply();
        }, null, "Error while getting user details, Please try again.", $scope);
    }

    $scope.UpdateUserAccount = function () {
        if ($scope.username == "" || $scope.username == null) {
            objCommon.ShowMessage("Username is required", "error");
        }
        else if ($scope.email == "" || $scope.email == null) {
            objCommon.ShowMessage("Email is required", "error");
        }
        else if ($scope.mobile == "" || $scope.mobile == null) {
            objCommon.ShowMessage("Mobile number is required", "error");
        }
        else if ($scope.address == "" || $scope.address == null) {
            objCommon.ShowMessage("Address is required", "error");
        }
        else {
            $("body").css({ "opacity": "0.5" });
            $scope.loader = true;
            var data = {
                username: $scope.username,
                email: $scope.email,
                mobile: $scope.mobile,
                address: $scope.address,
            };
            objCommon.AjaxCall("User/UpdateUserDetail", $.param(data), "GET", true, function (response) {
                $scope.loader = false;
                $("body").css({ "opacity": "1" });
                if (response.Status == true) {
                    objCommon.ShowMessage(response.RetMessage, "success");
                }
                else {
                    objCommon.ShowMessage(response.RetMessage, "error");
                }
                $scope.$apply();
            }, null, "Error while updating user details, Please try again.", $scope);
        }  
    };

    $scope.AddressesList = function () {
        $("body").css({ "opacity": "0.5" });
        $scope.loader = true;
        var data = {};
        objCommon.AjaxCall("User/GetAddressList", $.param(data), "GET", true, function (data) {
            if (data == "Invalid") {
                objCommon.ShowMessage("Access Denied Your Session is Time Out.", "error");
            }
            else {
                $scope.AddressList = data;
            }
            $("body").css({ "opacity": "1" });
            $scope.loader = false;
            $scope.$apply();
        }, null, "Error while getting address list, Please try again.", $scope)

    }

    $scope.ChangePassword = function ($event) {
        if ($scope.OldPassword == "" || $scope.OldPassword == null) {
            objCommon.ShowMessage("Please enter old password", "error");
        }
        else if ($scope.NewPassword == "" || $scope.NewPassword == null) {
            objCommon.ShowMessage("Please enter new password", "error");
        }
        else if ($scope.ConfirmPassword == "" || $scope.ConfirmPassword == null) {
            objCommon.ShowMessage("Please enter confirm password", "error");
        }
        else if ($scope.NewPassword != $scope.ConfirmPassword) {
            objCommon.ShowMessage("New password and confirm password not matched", "error");
        }
        else {
            $("body").css({ "opacity": "0.5" });
            $scope.loader = true;
            var data = {
                OldPassword: $scope.OldPassword,
                NewPassword: $scope.NewPassword
            };
            objCommon.AjaxCall("User/ChangePassword", $.param(data), "GET", true, function (response) {
                $("body").css({ "opacity": "1" });
                $scope.loader = false;
                if (response.Status == true) {
                    objCommon.ShowMessage(response.RetMessage, "success");
                    window.location.replace(objCommon.baseUrl + "Accounts/SignIn");
                }
                else {
                    objCommon.ShowMessage(response.RetMessage, "error");
                }
                $scope.$apply();
            }, null, "Error while changing password, Please try again.", $scope);
        }
    };

    $scope.SignOut = function () {
        objCommon.AjaxCall("Accounts/SignOut", $.param({}), "GET", true, function (d) {
            if (d == "success") {
                window.location.replace(objCommon.baseUrl + "Home/Index");
            }
            else {
                objCommon.ShowMessage(d, "error");
            }
        }, null, "Error while sign out user, Please try again.", $scope);
    }

});
